src/components/InLineNotification.tsx

```diff
@@ -0,0 +1,158 @@
+import React from 'react'
+import { StyleProp, StyleSheet, Text, TextStyle, View, ViewStyle } from 'react-native'
+import AttentionIcon from 'src/icons/Attention'
+import Colors from 'src/styles/colors'
+import fontStyles, { typeScale } from 'src/styles/fonts'
+import { Spacing } from 'src/styles/styles'
+
+export enum Severity {
+  Informational,
+  Warning,
+  Error,
+}
+
+interface Props {
+  severity: Severity
+  title?: string | null
+  description: string
+  style?: StyleProp<ViewStyle>
+  ctaLabel?: string | null
+  onPressCta?: () => void
+  ctaLabel2?: string | null
+  onPressCta2?: () => void
+  testID?: string
+}
+
+interface CustomStyles {
+  container: ViewStyle
+  attentionIcon: Record<'color', Colors>
+  ctaLabel: TextStyle
+  ctaLabel2: TextStyle
+}
+
+export function InLineNotification({
+  severity,
+  title,
+  description,
+  style,
+  ctaLabel,
+  onPressCta,
+  ctaLabel2,
+  onPressCta2,
+  testID,
+}: Props) {
+  const severityStyle = severityStyles[severity]
+  return (
+    <View style={[defaultStyles.container, severityStyle.container, style]} testID={testID}>
+      {title && (
+        <View style={defaultStyles.row}>
+          <View style={defaultStyles.attentionIcon}>
+            <AttentionIcon color={severityStyle.attentionIcon.color} />
+          </View>
+          <Text style={defaultStyles.titleText}>{title}</Text>
+        </View>
+      )}
+
+      <View style={defaultStyles.bodyContainer}>
+        <View style={[defaultStyles.row, defaultStyles.bodyRow]}>
+          <View style={defaultStyles.attentionIcon}>
+            {!title && <AttentionIcon color={severityStyle.attentionIcon.color} />}
+          </View>
+          <Text style={defaultStyles.bodyText}>{description}</Text>
+        </View>
```

Comment:
i think that the structure can be simplified - the way you have it here, you need to declare the icon twice. but it seems like the icon should always be there. to simplify you could have a structure like
```
<View style={style.row}>
  <Icon>
  <View>
    {title ?? <Text {...} />}
    {description}
    <View style={styles.row}>
      {ctas}
    </View>
  </View>
</View>
```
alternatively you could have a row that has the icon on the left, title and description on the right, and a second row for buttons

Reply:
I've changed this part a bit to define the icon just once and remove some redundant View components.

I've went with a solution where the title and description are in separate rows. I chose that because we vertically center the icon with the title.